### PR TITLE
Lwm2m2 unitest

### PR DESCRIPTION
**Pull request by beingStrongeryqqq**
**What this PR does / why we need it**:
Make unitest for ciphersuite.go and deviceshifulwm2mconfig.go
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [x] unit test
- [ ] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/deviceshifu/deviceshifulwm2m/lwm2m/ciphersuite_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	for _, cipherSuite := range cipherSuiteStrs {
+		res, err := CipherSuiteStringToCode(cipherSuite)
+		if err != nil {
+			t.Errorf("unknown cipher suite: %v", err)
+		}
+		if res != cipherSuiteMap[cipherSuite] {
+			t.Errorf("Error in mapping cipher suite: %v", cipherSuite)
+		}
+	}
+}
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/lwm2m/ciphersuite_test.go:54]: Need to add test cases to verify that the string is not a cipher suite string or not supported in the cipher suite map.

File: pkg/deviceshifu/deviceshifulwm2m/lwm2m/ciphersuite_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA:      true,
+	TLS_PSK_WITH_AES_128_CCM:                true,
+	TLS_PSK_WITH_AES_128_CCM_8:              true,
+	TLS_PSK_WITH_AES_256_CCM_8:              true,
+	TLS_PSK_WITH_AES_128_GCM_SHA256:         true,
+	TLS_PSK_WITH_AES_128_CBC_SHA256:         true,
+	TLS_ECDHE_PSK_WITH_AES_128_CBC_SHA256:   true,
+}
+
+func TestCipherSuiteStringsToCodes(t *testing.T) {
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/lwm2m/ciphersuite_test.go:73]: ditto

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifubase"
+)
+
+func TestCreateLwM2MInstructions(t *testing.T) {
+	tests := []struct {
+		InstructionName string
+		input           *deviceshifubase.DeviceShifuInstructions
+		expected        *LwM2MInstruction
+		expectingPanic  bool
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:14]: when will except panic?

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+					"instruction": {
+						ObjectId:      "1",
+						EnableObserve: false,
+					},
+				},
+			},
+			expectingPanic: false,
+		},
+	}
+
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:61]: only success test case? need to add some failed test cases

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+			// CreateLwM2MInstructions's logger.Fatalf will end program with no panic
+			res := CreateLwM2MInstructions(tt.input)
+			if !tt.expectingPanic && !compareLwM2MInstructions(res, tt.expected) {
+				t.Errorf("Test case %s failed: expected %v, got %v", tt.InstructionName, tt.expected, res)
+			}
+		})
+	}
+}
+
+func compareLwM2MInstructions(a, b *LwM2MInstruction) bool {
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:73]: can we use `DeepEqual` to do it?

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+import (
+	"testing"
+
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifubase"
+)
+
+func TestCreateLwM2MInstructions(t *testing.T) {
+	tests := []struct {
+		InstructionName string
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:11]: TestCase name or just Name

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go (hunk 3 above)
Comment (beingStrongeryqqq) [reply pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go]: judge whether the returned result is panic

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifubase"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateLwM2MInstructions(t *testing.T) {
+	tests := []struct {
+		name           string
+		input          *deviceshifubase.DeviceShifuInstructions
+		expected       *LwM2MInstruction
+		expectingPanic bool
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:16]: when will except panic? It seems that you never use this field or expect is true

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.expectingPanic {
+				assert.Panics(t, func() { CreateLwM2MInstructions(tt.input) })
+			} else {
+				result := CreateLwM2MInstructions(tt.input)
+				if !reflect.DeepEqual(tt.expected, result) {
+					t.Errorf("expected %v, got %v", tt.expected, result)
+				}
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:50]: can we use assert.ObjectAreEqual to instead?

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+func TestCreateLwM2MInstructions(t *testing.T) {
+	tests := []struct {
+		name           string
+		input          *deviceshifubase.DeviceShifuInstructions
+		expected       *LwM2MInstruction
+		expectingPanic bool
+	}{
+		{
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:18]: add some test cases:
- instruction map is empty
- duplicate object id in different instruction
- objectId is empty or enableObserStr is empty(if we can test it)

Comment (beingStrongeryqqq) [reply pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:18]: > add some test cases:
> 
> * instruction map is empty
> * duplicate object id in different instruction
> * objectId is empty or enableObserStr is empty(if we can test it)

The source code uses fatalf, which will terminate the program directly.

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+	// Check if each instruction's properties are correctly mapped
+	instruction1 := result.Instructions["instruction1"]
+	assert.NotNil(t, instruction1)
+	assert.Equal(t, "123", instruction1.ObjectId)
+	assert.True(t, instruction1.EnableObserve)
+
+	instruction2 := result.Instructions["instruction2"]
+	assert.NotNil(t, instruction2)
+	assert.Equal(t, "456", instruction2.ObjectId)
+	assert.False(t, instruction2.EnableObserve)
```
Comment (tomqin93) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2mconfig_test.go:45]: use equal instead of comparing each value

**Pull request closed without merging**